lab_6_pipeline/pipeline.py

```diff
@@ -18,21 +32,46 @@ def __init__(self, path_to_raw_txt_data: Path):
         """
         Initializes CorpusManager
         """
+        self.path_to_raw_text_data = path_to_raw_txt_data
+        self._storage = {}
+        self._validate_dataset()
+        self._scan_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validates folder with assets
         """
+        if not self.path_to_raw_text_data.exists():
+            raise FileNotFoundError
+        if not self.path_to_raw_text_data.is_dir():
+            raise NotADirectoryError
+        if not len(list(self.path_to_raw_text_data.glob('*.*'))):
+            raise EmptyDirectoryError
+        raw_paths = list(self.path_to_raw_text_data.glob('*_raw.txt'))
+        meta_paths = list(self.path_to_raw_text_data.glob('*_meta.json'))
+        if len(raw_paths) != len(meta_paths):
+            raise InconsistentDatasetError
+        for i in range(1, len(raw_paths) + 1):
```

Comment:
too straightforward and brutal, you can do it better

Reply:
I know I can do it better, and this knowledge really makes me proud of myself. Not going to implement it though.